projects/bootloader/src/config.c

```diff
@@ -0,0 +1,87 @@
+#include "config.h"
+#include "crc32.h"
+#include "flash.h"
+#include "log.h"
+#include "persist.h"
+#include "soft_timer.h"
+
+static PersistStorage s_config_1_persist = { 0 };
+static BootloaderConfig s_config_1_blob = { 0 };
+
+static PersistStorage s_config_2_persist = { 0 };
+static BootloaderConfig s_config_2_blob = { 0 };
+
+#define BOOTLOADER_CONFIG_PAGE_1_FLASH_PAGE (FLASH_ADDR_TO_PAGE(BOOTLOADER_CONFIG_PAGE_1_START))
+#define BOOTLOADER_CONFIG_PAGE_2_FLASH_PAGE (FLASH_ADDR_TO_PAGE(BOOTLOADER_CONFIG_PAGE_2_START))
+
+StatusCode config_init(void) {
+  status_ok_or_return(persist_init(&s_config_1_persist, BOOTLOADER_CONFIG_PAGE_1_FLASH_PAGE,
+                                   &s_config_1_blob, sizeof(s_config_1_blob), false));
+  status_ok_or_return(persist_ctrl_periodic(&s_config_1_persist, false));
+
+  status_ok_or_return(persist_init(&s_config_2_persist, BOOTLOADER_CONFIG_PAGE_2_FLASH_PAGE,
+                                   &s_config_2_blob, sizeof(s_config_2_blob), false));
+  status_ok_or_return(persist_ctrl_periodic(&s_config_2_persist, false));
+
+  uint32_t config_1_crc = s_config_1_blob.CRC32;
+  s_config_1_blob.CRC32 = 0;
+
+  uint32_t config_2_crc = s_config_2_blob.CRC32;
+  s_config_2_blob.CRC32 = 0;
+
+  uint32_t config_1_check_crc = crc32_arr((uint8_t *)&s_config_1_blob, sizeof(s_config_1_blob));
+  uint32_t config_2_check_crc = crc32_arr((uint8_t *)&s_config_2_blob, sizeof(s_config_2_blob));
+
+  bool is_config_1_equal = config_1_crc == config_1_check_crc;
+  bool is_config_2_equal = config_2_crc == config_2_check_crc;
```

Comment:
`is_config_X_equal` isn't a particularly descriptive name - if I just see the name, what does it mean for one of the configs to be "equal"? Change these names to be something about whether it's ok or corrupted; IMO the most readable form would be to invert them and name them something like `is_config_X_corrupted`.